gpt_index/indices/query/knowledge_graph/query.py

```diff
@@ -37,17 +41,20 @@ def __init__(
         self,
         index_struct: KG,
         query_keyword_extract_template: Optional[QueryKeywordExtractPrompt] = None,
+        embed_model: Optional[BaseEmbedding] = None,
```

Comment:
i have a high-level suggestion to make. i think right now the "mode" during index construction and querying are coupled, and it would be worth decoupling them a bit. 

More concretely, during index construction I think we can have an `include_embeddings` boolean, where if it's true, then we add embeddings through `index_struct.add_to_embedding_dict`. This boolean doesn't need to be stored on the index struct, just as a variable in the outer index class.

During query-time, the user may want to specify a keyword-based query or an embedding-based query or a keyword+embedding-based query, whether or not the index struct has embeddings (though obv embedding-based queries should fail if the index struct doesn't have embeddings). We can add that as a parameter here for instance named `embedding_mode` which can be None (keyword-only), "embedding_only", or "hybrid".

You can also set a default for `embedding_mode` based on whether `include_embeddings` is true (see overriding _preprocess_query for how to do this).

What's the benefit of this? Allowing users flexibility in queries even if they do choose to create the KG with embeddings (and we could argue we always want to create the KG with embeddings)

Reply:
Done!